process/http.go

```diff
@@ -71,37 +75,56 @@ func (p procHTTP) Close(context.Context) error {
 	return nil
 }
 
-// Batch processes one or more capsules with the processor. Conditions are
-// optionally applied to the data to enable processing.
-func (p procHTTP) Batch(ctx context.Context, capsules ...config.Capsule) ([]config.Capsule, error) {
-	return batchApply(ctx, capsules, p, p.Condition)
-}
+// Create a new HTTP processor.
+func newProcHTTP(cfg config.Config) (p procHTTP, err error) {
+	err = config.Decode(cfg.Settings, &p)
+	if err != nil {
+		return procHTTP{}, err
+	}
+
+	p.operator, err = condition.NewOperator(p.Condition)
+	if err != nil {
+		return procHTTP{}, err
+	}
 
-// Apply processes a capsule with the processor.
-func (p procHTTP) Apply(ctx context.Context, capsule config.Capsule) (config.Capsule, error) {
 	// error early if required options are missing
 	if p.Options.URL == "" {
-		return capsule, fmt.Errorf("process: http: options %+v: %v", p.Options, errMissingRequiredOptions)
+		return procHTTP{}, fmt.Errorf("process: http: option url: %v", errors.ErrMissingRequiredOption)
+	}
+
+	if p.Options.Method == "POST" && p.Options.BodyKey == "" {
+		return procHTTP{}, fmt.Errorf("process: http: options body_key: %v", errors.ErrMissingRequiredOption)
 	}
 
 	if !httpClient.IsEnabled() {
 		httpClient.Setup()
 	}
 
-	var headers []http.Header
+	ctx := context.Background()
```

Comment:
refactor: we should pass the context from the source, this would require a breaking change to the public New* functions. in this case, secrets.Interpolate calls secrets.Get which may call the AWS Secrets Manager.